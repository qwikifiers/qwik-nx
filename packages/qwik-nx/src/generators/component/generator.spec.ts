import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';
import { Tree } from '@nrwl/devkit';
import componentGenerator from './generator';
import {
  createLib,
  getFormattedListChanges,
} from './../../utils/testing-generators';

describe('component generator', () => {
  let appTree: Tree;
  const projectName = 'dummy-lib';

  beforeEach(() => {
    appTree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
    createLib(appTree, projectName);
  });

  it('should generate a component file inside a given directory', async () => {
    await componentGenerator(appTree, {
      name: 'hello',
      project: projectName,
    });

    expect(getFormattedListChanges(appTree)).toMatchSnapshot();
  });

  describe('generated contents should match the snapshot', () => {
    // [ '000', '001', '010', '011', '100', '101', '110', '111' ]
    test.each`
      exportDefault | generateStories | skipTests
      ${false}      | ${false}        | ${false}
      ${false}      | ${false}        | ${true}
      ${false}      | ${true}         | ${false}
      ${false}      | ${true}         | ${true}
      ${true}       | ${false}        | ${false}
      ${true}       | ${false}        | ${true}
      ${true}       | ${true}         | ${false}
      ${true}       | ${true}         | ${true}
    `(
      'should match the snapshot when exportDefault is "$exportDefault", generateStories is "$generateStories" and skipTests is "$skipTests" ',
      async ({ exportDefault, generateStories, skipTests }) => {
        await componentGenerator(appTree, {
          name: 'hello',
          project: projectName,
          generateStories,
          skipTests,
          exportDefault,
        });
        expect(getFormattedListChanges(appTree)).toMatchSnapshot();
        expect(
          appTree
            .read(`libs/${projectName}/src/lib/hello/hello.tsx`)
            ?.toString()
        ).toMatchSnapshot('hello.tsx');
        expect(
          appTree
            .read(`libs/${projectName}/src/lib/hello/hello.spec.tsx`)
            ?.toString()
        ).toMatchSnapshot('hello.spec.tsx');
        expect(
          appTree
            .read(`libs/${projectName}/src/lib/hello/hello.stories.tsx`)
            ?.toString()
        ).toMatchSnapshot('hello.stories.tsx');
        expect(
          appTree
            .read(`libs/${projectName}/src/lib/hello/hello.doc.mdx`)
            ?.toString()
        ).toMatchSnapshot('hello.doc.mdx');
      }
    );
  });
});
