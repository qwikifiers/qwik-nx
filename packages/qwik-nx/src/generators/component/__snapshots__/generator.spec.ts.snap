// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "false", generateStories is "false" and skipTests is "false"  1`] = `
Array [
  Object {
    "path": ".prettierrc",
    "type": "CREATE",
  },
  Object {
    "path": "apps/.gitignore",
    "type": "CREATE",
  },
  Object {
    "path": "libs/.gitignore",
    "type": "CREATE",
  },
  Object {
    "path": "libs/dummy-lib/project.json",
    "type": "CREATE",
  },
  Object {
    "path": "libs/dummy-lib/src/lib/hello/hello.spec.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "libs/dummy-lib/src/lib/hello/hello.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "nx.json",
    "type": "CREATE",
  },
  Object {
    "path": "package.json",
    "type": "CREATE",
  },
  Object {
    "path": "tsconfig.base.json",
    "type": "CREATE",
  },
]
`;

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "false", generateStories is "false" and skipTests is "false" : hello.doc.mdx 1`] = `undefined`;

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "false", generateStories is "false" and skipTests is "false" : hello.spec.tsx 1`] = `
"import { createDOM } from '@builder.io/qwik/testing';
import { test, expect } from 'vitest';
import { Hello } from './hello';

test(\`[Hello Component]: Should render\`, async () => {
  const { screen, render } = await createDOM();
  await render(<Hello />);
  expect(screen.innerHTML).toContain('Hello works!');
});
"
`;

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "false", generateStories is "false" and skipTests is "false" : hello.stories.tsx 1`] = `undefined`;

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "false", generateStories is "false" and skipTests is "false" : hello.tsx 1`] = `
"import { component$ } from '@builder.io/qwik';

export const Hello = component$(() => {
  return <>Hello works!</>;
});
"
`;

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "false", generateStories is "false" and skipTests is "true"  1`] = `
Array [
  Object {
    "path": ".prettierrc",
    "type": "CREATE",
  },
  Object {
    "path": "apps/.gitignore",
    "type": "CREATE",
  },
  Object {
    "path": "libs/.gitignore",
    "type": "CREATE",
  },
  Object {
    "path": "libs/dummy-lib/project.json",
    "type": "CREATE",
  },
  Object {
    "path": "libs/dummy-lib/src/lib/hello/hello.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "nx.json",
    "type": "CREATE",
  },
  Object {
    "path": "package.json",
    "type": "CREATE",
  },
  Object {
    "path": "tsconfig.base.json",
    "type": "CREATE",
  },
]
`;

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "false", generateStories is "false" and skipTests is "true" : hello.doc.mdx 1`] = `undefined`;

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "false", generateStories is "false" and skipTests is "true" : hello.spec.tsx 1`] = `undefined`;

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "false", generateStories is "false" and skipTests is "true" : hello.stories.tsx 1`] = `undefined`;

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "false", generateStories is "false" and skipTests is "true" : hello.tsx 1`] = `
"import { component$ } from '@builder.io/qwik';

export const Hello = component$(() => {
  return <>Hello works!</>;
});
"
`;

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "false", generateStories is "true" and skipTests is "false"  1`] = `
Array [
  Object {
    "path": ".prettierrc",
    "type": "CREATE",
  },
  Object {
    "path": "apps/.gitignore",
    "type": "CREATE",
  },
  Object {
    "path": "libs/.gitignore",
    "type": "CREATE",
  },
  Object {
    "path": "libs/dummy-lib/project.json",
    "type": "CREATE",
  },
  Object {
    "path": "libs/dummy-lib/src/lib/hello/hello.doc.mdx",
    "type": "CREATE",
  },
  Object {
    "path": "libs/dummy-lib/src/lib/hello/hello.spec.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "libs/dummy-lib/src/lib/hello/hello.stories.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "libs/dummy-lib/src/lib/hello/hello.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "nx.json",
    "type": "CREATE",
  },
  Object {
    "path": "package.json",
    "type": "CREATE",
  },
  Object {
    "path": "tsconfig.base.json",
    "type": "CREATE",
  },
]
`;

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "false", generateStories is "true" and skipTests is "false" : hello.doc.mdx 1`] = `
"import { Canvas, Story } from '@storybook/addon-docs';
import { Hello } from './hello';

# Hello Component

## Purpose

{/* Why the component is needed */}

## Example

{/* Common copy/paste example that people can throw into their templates and ts */}

\`\`\`tsx
<Hello param=\\"value\\" />
\`\`\`

## Use case examples

{/* Examples based on use cases */}

### Primary

<Canvas>
  <Story id=\\"hello--primary\\" />
</Canvas>
"
`;

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "false", generateStories is "true" and skipTests is "false" : hello.spec.tsx 1`] = `
"import { createDOM } from '@builder.io/qwik/testing';
import { test, expect } from 'vitest';
import { Hello } from './hello';

test(\`[Hello Component]: Should render\`, async () => {
  const { screen, render } = await createDOM();
  await render(<Hello />);
  expect(screen.innerHTML).toContain('Hello works!');
});
"
`;

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "false", generateStories is "true" and skipTests is "false" : hello.stories.tsx 1`] = `
"import type { Meta } from 'storybook-framework-qwik';
import { Hello } from './hello';
import doc from './hello.doc.mdx';

export default {
  title: 'Hello',
  tags: ['autodocs'],
  parameters: {
    docs: {
      page: doc,
    },
  },
  argTypes: {
    // put component params here
  },
  render(args) {
    return <Hello {...args} />;
  },
} as Meta;

export const Primary = {};
"
`;

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "false", generateStories is "true" and skipTests is "false" : hello.tsx 1`] = `
"import { component$ } from '@builder.io/qwik';

export const Hello = component$(() => {
  return <>Hello works!</>;
});
"
`;

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "false", generateStories is "true" and skipTests is "true"  1`] = `
Array [
  Object {
    "path": ".prettierrc",
    "type": "CREATE",
  },
  Object {
    "path": "apps/.gitignore",
    "type": "CREATE",
  },
  Object {
    "path": "libs/.gitignore",
    "type": "CREATE",
  },
  Object {
    "path": "libs/dummy-lib/project.json",
    "type": "CREATE",
  },
  Object {
    "path": "libs/dummy-lib/src/lib/hello/hello.doc.mdx",
    "type": "CREATE",
  },
  Object {
    "path": "libs/dummy-lib/src/lib/hello/hello.stories.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "libs/dummy-lib/src/lib/hello/hello.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "nx.json",
    "type": "CREATE",
  },
  Object {
    "path": "package.json",
    "type": "CREATE",
  },
  Object {
    "path": "tsconfig.base.json",
    "type": "CREATE",
  },
]
`;

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "false", generateStories is "true" and skipTests is "true" : hello.doc.mdx 1`] = `
"import { Canvas, Story } from '@storybook/addon-docs';
import { Hello } from './hello';

# Hello Component

## Purpose

{/* Why the component is needed */}

## Example

{/* Common copy/paste example that people can throw into their templates and ts */}

\`\`\`tsx
<Hello param=\\"value\\" />
\`\`\`

## Use case examples

{/* Examples based on use cases */}

### Primary

<Canvas>
  <Story id=\\"hello--primary\\" />
</Canvas>
"
`;

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "false", generateStories is "true" and skipTests is "true" : hello.spec.tsx 1`] = `undefined`;

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "false", generateStories is "true" and skipTests is "true" : hello.stories.tsx 1`] = `
"import type { Meta } from 'storybook-framework-qwik';
import { Hello } from './hello';
import doc from './hello.doc.mdx';

export default {
  title: 'Hello',
  tags: ['autodocs'],
  parameters: {
    docs: {
      page: doc,
    },
  },
  argTypes: {
    // put component params here
  },
  render(args) {
    return <Hello {...args} />;
  },
} as Meta;

export const Primary = {};
"
`;

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "false", generateStories is "true" and skipTests is "true" : hello.tsx 1`] = `
"import { component$ } from '@builder.io/qwik';

export const Hello = component$(() => {
  return <>Hello works!</>;
});
"
`;

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "true", generateStories is "false" and skipTests is "false"  1`] = `
Array [
  Object {
    "path": ".prettierrc",
    "type": "CREATE",
  },
  Object {
    "path": "apps/.gitignore",
    "type": "CREATE",
  },
  Object {
    "path": "libs/.gitignore",
    "type": "CREATE",
  },
  Object {
    "path": "libs/dummy-lib/project.json",
    "type": "CREATE",
  },
  Object {
    "path": "libs/dummy-lib/src/lib/hello/hello.spec.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "libs/dummy-lib/src/lib/hello/hello.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "nx.json",
    "type": "CREATE",
  },
  Object {
    "path": "package.json",
    "type": "CREATE",
  },
  Object {
    "path": "tsconfig.base.json",
    "type": "CREATE",
  },
]
`;

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "true", generateStories is "false" and skipTests is "false" : hello.doc.mdx 1`] = `undefined`;

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "true", generateStories is "false" and skipTests is "false" : hello.spec.tsx 1`] = `
"import { createDOM } from '@builder.io/qwik/testing';
import { test, expect } from 'vitest';
import Hello from './hello';

test(\`[Hello Component]: Should render\`, async () => {
  const { screen, render } = await createDOM();
  await render(<Hello />);
  expect(screen.innerHTML).toContain('Hello works!');
});
"
`;

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "true", generateStories is "false" and skipTests is "false" : hello.stories.tsx 1`] = `undefined`;

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "true", generateStories is "false" and skipTests is "false" : hello.tsx 1`] = `
"import { component$ } from '@builder.io/qwik';

export default component$(() => {
  return <>Hello works!</>;
});
"
`;

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "true", generateStories is "false" and skipTests is "true"  1`] = `
Array [
  Object {
    "path": ".prettierrc",
    "type": "CREATE",
  },
  Object {
    "path": "apps/.gitignore",
    "type": "CREATE",
  },
  Object {
    "path": "libs/.gitignore",
    "type": "CREATE",
  },
  Object {
    "path": "libs/dummy-lib/project.json",
    "type": "CREATE",
  },
  Object {
    "path": "libs/dummy-lib/src/lib/hello/hello.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "nx.json",
    "type": "CREATE",
  },
  Object {
    "path": "package.json",
    "type": "CREATE",
  },
  Object {
    "path": "tsconfig.base.json",
    "type": "CREATE",
  },
]
`;

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "true", generateStories is "false" and skipTests is "true" : hello.doc.mdx 1`] = `undefined`;

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "true", generateStories is "false" and skipTests is "true" : hello.spec.tsx 1`] = `undefined`;

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "true", generateStories is "false" and skipTests is "true" : hello.stories.tsx 1`] = `undefined`;

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "true", generateStories is "false" and skipTests is "true" : hello.tsx 1`] = `
"import { component$ } from '@builder.io/qwik';

export default component$(() => {
  return <>Hello works!</>;
});
"
`;

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "true", generateStories is "true" and skipTests is "false"  1`] = `
Array [
  Object {
    "path": ".prettierrc",
    "type": "CREATE",
  },
  Object {
    "path": "apps/.gitignore",
    "type": "CREATE",
  },
  Object {
    "path": "libs/.gitignore",
    "type": "CREATE",
  },
  Object {
    "path": "libs/dummy-lib/project.json",
    "type": "CREATE",
  },
  Object {
    "path": "libs/dummy-lib/src/lib/hello/hello.doc.mdx",
    "type": "CREATE",
  },
  Object {
    "path": "libs/dummy-lib/src/lib/hello/hello.spec.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "libs/dummy-lib/src/lib/hello/hello.stories.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "libs/dummy-lib/src/lib/hello/hello.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "nx.json",
    "type": "CREATE",
  },
  Object {
    "path": "package.json",
    "type": "CREATE",
  },
  Object {
    "path": "tsconfig.base.json",
    "type": "CREATE",
  },
]
`;

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "true", generateStories is "true" and skipTests is "false" : hello.doc.mdx 1`] = `
"import { Canvas, Story } from '@storybook/addon-docs';
import Hello from './hello';

# Hello Component

## Purpose

{/* Why the component is needed */}

## Example

{/* Common copy/paste example that people can throw into their templates and ts */}

\`\`\`tsx
<Hello param=\\"value\\" />
\`\`\`

## Use case examples

{/* Examples based on use cases */}

### Primary

<Canvas>
  <Story id=\\"hello--primary\\" />
</Canvas>
"
`;

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "true", generateStories is "true" and skipTests is "false" : hello.spec.tsx 1`] = `
"import { createDOM } from '@builder.io/qwik/testing';
import { test, expect } from 'vitest';
import Hello from './hello';

test(\`[Hello Component]: Should render\`, async () => {
  const { screen, render } = await createDOM();
  await render(<Hello />);
  expect(screen.innerHTML).toContain('Hello works!');
});
"
`;

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "true", generateStories is "true" and skipTests is "false" : hello.stories.tsx 1`] = `
"import type { Meta } from 'storybook-framework-qwik';
import Hello from './hello';
import doc from './hello.doc.mdx';

export default {
  title: 'Hello',
  tags: ['autodocs'],
  parameters: {
    docs: {
      page: doc,
    },
  },
  argTypes: {
    // put component params here
  },
  render(args) {
    return <Hello {...args} />;
  },
} as Meta;

export const Primary = {};
"
`;

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "true", generateStories is "true" and skipTests is "false" : hello.tsx 1`] = `
"import { component$ } from '@builder.io/qwik';

export default component$(() => {
  return <>Hello works!</>;
});
"
`;

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "true", generateStories is "true" and skipTests is "true"  1`] = `
Array [
  Object {
    "path": ".prettierrc",
    "type": "CREATE",
  },
  Object {
    "path": "apps/.gitignore",
    "type": "CREATE",
  },
  Object {
    "path": "libs/.gitignore",
    "type": "CREATE",
  },
  Object {
    "path": "libs/dummy-lib/project.json",
    "type": "CREATE",
  },
  Object {
    "path": "libs/dummy-lib/src/lib/hello/hello.doc.mdx",
    "type": "CREATE",
  },
  Object {
    "path": "libs/dummy-lib/src/lib/hello/hello.stories.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "libs/dummy-lib/src/lib/hello/hello.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "nx.json",
    "type": "CREATE",
  },
  Object {
    "path": "package.json",
    "type": "CREATE",
  },
  Object {
    "path": "tsconfig.base.json",
    "type": "CREATE",
  },
]
`;

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "true", generateStories is "true" and skipTests is "true" : hello.doc.mdx 1`] = `
"import { Canvas, Story } from '@storybook/addon-docs';
import Hello from './hello';

# Hello Component

## Purpose

{/* Why the component is needed */}

## Example

{/* Common copy/paste example that people can throw into their templates and ts */}

\`\`\`tsx
<Hello param=\\"value\\" />
\`\`\`

## Use case examples

{/* Examples based on use cases */}

### Primary

<Canvas>
  <Story id=\\"hello--primary\\" />
</Canvas>
"
`;

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "true", generateStories is "true" and skipTests is "true" : hello.spec.tsx 1`] = `undefined`;

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "true", generateStories is "true" and skipTests is "true" : hello.stories.tsx 1`] = `
"import type { Meta } from 'storybook-framework-qwik';
import Hello from './hello';
import doc from './hello.doc.mdx';

export default {
  title: 'Hello',
  tags: ['autodocs'],
  parameters: {
    docs: {
      page: doc,
    },
  },
  argTypes: {
    // put component params here
  },
  render(args) {
    return <Hello {...args} />;
  },
} as Meta;

export const Primary = {};
"
`;

exports[`component generator generated contents should match the snapshot should match the snapshot when exportDefault is "true", generateStories is "true" and skipTests is "true" : hello.tsx 1`] = `
"import { component$ } from '@builder.io/qwik';

export default component$(() => {
  return <>Hello works!</>;
});
"
`;

exports[`component generator should generate a component file inside a given directory 1`] = `
Array [
  Object {
    "path": ".prettierrc",
    "type": "CREATE",
  },
  Object {
    "path": "apps/.gitignore",
    "type": "CREATE",
  },
  Object {
    "path": "libs/.gitignore",
    "type": "CREATE",
  },
  Object {
    "path": "libs/dummy-lib/project.json",
    "type": "CREATE",
  },
  Object {
    "path": "libs/dummy-lib/src/lib/hello/hello.spec.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "libs/dummy-lib/src/lib/hello/hello.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "nx.json",
    "type": "CREATE",
  },
  Object {
    "path": "package.json",
    "type": "CREATE",
  },
  Object {
    "path": "tsconfig.base.json",
    "type": "CREATE",
  },
]
`;
